### ISSUE TITLE
Bug: no-fallthrough interaction with disable comments and "unused disable directive" warnings

### ISSUE THREAD

**Issue by bradzacher** [eslint/eslint#16650]
### Environment

ESLint version: 8.29.0

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
module.exports = {
  reportUnusedDisableDirectives: true,
  rules: {
    'no-fallthrough': 'error',
  },
};
```
</details>

```js
switch (test) {
  case a:
    hello();

  // eslint-disable-next-line no-fallthrough
  case b:
}
```

### What did you expect to happen?

The rule report should be hidden and no "unused disable directive" warnings should be reported.

### What actually happened?

An unused disable directive warning is reported
![image](https://user-images.githubusercontent.com/7462525/206941185-9187b66c-f44c-48e3-9df5-047f7fa2251a.png)

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

This default behaviour is problematic because if you run `eslint --fix`, then eslint will remove the unused disable comment, causing no-fallthrough error.

I think that this occurs because the default comment regex will match the `no-fallthrough` string
https://github.com/eslint/eslint/blob/ad44344ef6fdeac7217eb83bc54a230382c0da5e/lib/rules/no-fallthrough.js#L11

IMO either the rule should ignore eslint-disable directives, or the default regex should be updated to ignore them to prevent this behaviour.

**Comment by bradzacher** [comment]
It seems like a weird inconsistency in a core rule that you shouldn't disable.

I wouldn't mind if it was something like `eslint-plugin-eslint-comments` reporting on the unused disable - but the fact that eslint itself reports on the unused disable and subsequently "fix deletes" the valid comment to then cause the lint error that the comment was suppressing seems like really broken behaviour.

Esp considering the default for `reportUnusedDisableDirectives` is currently false, but in future it might be true - meaning with the default config eslint would delete a valid disable comment!

-----

For context what I was doing was enabling the rule on a large codebase. I don't have enough context for each of the violations to be able to determine if it's intentional or not so the workflow I've been using is to fix the ones I can, and those I can't add a disable comment like `// eslint-disable-next-line no-fallthrough -- TODO FIXME please review this fallthrough and determine if it was intentional` to explicitly mark the problem area for clean up.

But ofc my comment includes `fallthrough` (twice) which by default stops the rule from reporting, which triggers the bug and causes the disable to be marked as unused and get deleted.

**Comment by nzakas** [comment]
> IMO either the rule should ignore eslint-disable directives, or the default regex should be updated to ignore them to prevent this behaviour.

> but it might really be best that we update the rule to not treat eslint directives as fallthrough comments. 

I think this is the correct course of action here. This is an unfortunate naming collision that we should fix.

**Comment by gfyoung** [comment]
PR submitted to fix this! 🚀

Let me know what you all think.
